libbeat/autodiscover/providers/kubernetes/pod.go

```diff
@@ -271,6 +271,15 @@ func (p *pod) emit(pod *kubernetes.Pod, flag string) {
 
 	// Emit events for all initContainers
 	p.emitEvents(pod, flag, pod.Spec.InitContainers, pod.Status.InitContainerStatuses)
+
+	// Emit events for all ephemeralContainers
+	if p.config.ScrapeEphemeralContainers {
```

Comment:
I don't think we need to make this optional, let's always discover ephemeral containers.

But please add a comment here mentioning that this is an alpha feature in Kubernetes and we may need to remove or modify it if the feature is removed or modified in Kubernetes APIs.

Reply:
@jsoriano ok. I changed it as you requested